package/libnetwork/src/libnetwork.c

```diff
@@ -165,6 +171,65 @@ static size_t network_upload_read(char *buf, size_t size, size_t n, void *data)
   return -1;
 }
 
+static void trim_crlf(char* in_buf, char* out_buf, size_t buf_size)
+{
+  strncpy(out_buf, in_buf, buf_size);
+  char* crlf = strstr(out_buf, "\r\n");
+  *crlf = '\0';
+}
+
+static size_t network_upload_write(char *buf, size_t size, size_t n, void *data)
+{
+  network_config_t *cfg = (network_config_t*) data;
+
+  time_t now = time(NULL);
+
+  if (now - last_xfer_time < XFER_SECS) {
+    if (cfg->debug) {
+      piksi_log(LOG_DEBUG, "Last transfer too recent, pausing");
+    }
+    return CURL_READFUNC_PAUSE;
+  }
+
+  last_xfer_time = now;
+
+  FILE *fp_gga_cache = fopen("/persistent/nmea_GGA", "r");
```

Comment:
I don't love the `/persistent/nmea_GGA` vs. `/persistent/nmea/GGA`. Could have a `mkdir -p /persistent/nmea` in the daemon start.